File: Ex78ViewBinding/app/src/main/java/com/ch96/ex78viewbinding/MyAdapter.java
package com.ch96.ex78viewbinding;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.ch96.ex78viewbinding.databinding.RecyclerItemBinding;

import java.util.ArrayList;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.VH> {

    Context context;
    ArrayList<ItemVO> items;

    public MyAdapter(Context context, ArrayList<ItemVO> items) {
        this.context = context;
        this.items = items;
    }

    @NonNull
    @Override
    public VH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
       View itemView = LayoutInflater.from(context).inflate(R.layout.recycler_item, parent, false);
       return new VH(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull VH holder, int position) {
        holder.binding.tv.setText(items.get(position).title);
        holder.binding.iv.setImageResource(items.get(position).imgResId);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    class VH extends RecyclerView.ViewHolder {

        //recycler_item.xml과 연결되는 Binding 클래스 참조변수
        RecyclerItemBinding binding;

        public VH(@NonNull View itemView) {
            super(itemView);
            //이미 만든 뷰 객체의 안에 있는 자식 뷰들과 연결하는 Binding 객체
            binding = RecyclerItemBinding.bind(itemView); //inflate 이미 되어있음 bind만 하면 됨
        }
    }
}

File: Ex93KotlinOpenApiNaverSearch/app/src/main/java/com/ch96/ex93kotlinopenapinaversearch/MyAdapter.kt
package com.ch96.ex93kotlinopenapinaversearch

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.text.HtmlCompat
import androidx.recyclerview.widget.RecyclerView.Adapter
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.bumptech.glide.Glide
import com.ch96.ex93kotlinopenapinaversearch.databinding.RecyclerItemBinding

class MyAdapter (var context:Context, var items:MutableList<ShoppigItem>): Adapter<MyAdapter.VH>() { //상속받을때 인터페이스가 아니라면 () - 생성자 필수

    inner class VH(var binding: RecyclerItemBinding): ViewHolder(binding.root) //ViewHolder는 itemView를 VH로부터 받아야함 - binding 클래스 안에 root를 주자


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VH = VH(RecyclerItemBinding.inflate(LayoutInflater.from(context), parent, false))

    override fun getItemCount(): Int = items.size

    override fun onBindViewHolder(holder: VH, position: Int) {
        val item:ShoppigItem = items[position]

        //제목글씨중에 html태그문이 포함되어 있어서 지저분 - 제거
        var title:String = HtmlCompat.fromHtml(item.title, HtmlCompat.FROM_HTML_MODE_COMPACT).toString()
        holder.binding.tvTitle.text = title

        holder.binding.tvLowPrice.text = "${item.lprice}원"
        holder.binding.tvMall.text = item.mallName
        Glide.with(context).load(item.image).into(holder.binding.iv)

        holder.binding.root.setOnClickListener {
            //디바이스의 인터넷 앱을 실행
            val intent = Intent(Intent.ACTION_VIEW)
            intent.data = Uri.parse(item.link)
            context.startActivity(intent)
        }
    }
}